nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/FlowController.java

```diff
@@ -1156,11 +1214,14 @@ public void trigger(final ComponentNode component) {
                 }
             };
 
+            new TriggerFlowAnalysisTask(flowAnalyzer, rootProcessGroupSupplier).run();
             new TriggerValidationTask(flowManager, triggerIfValidating).run();
 
             final long millis = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - start);
             LOG.info("Performed initial validation of all components in {} milliseconds", millis);
 
+            // Trigger flow analysis to occur every 5 minutes.
+            flowAnalysisThreadPool.scheduleWithFixedDelay(new TriggerFlowAnalysisTask(flowManager.getFlowAnalyzer(), rootProcessGroupSupplier), 5, 5, TimeUnit.MINUTES);
```

Comment:
With the potential impact on performance, this interval should be promoted to a configurable value in nifi.properties.